### PR TITLE
Add tests for compilation of two source files with identical types.

### PR DESCRIPTION
**Pull request by kant2002**

### CODE REVIEW

**Comment by kant2002** [comment]
Probably depends on #481 since I was testing from that branch.

File: Cesium.CodeGen/Ir/Types/EnumType.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 namespace Cesium.CodeGen.Ir.Types;
 
-internal sealed class EnumType : IType
+internal sealed class EnumType : IType, IEquatable<EnumType>
```
Comment (ForNeVeR) [inline Cesium.CodeGen/Ir/Types/EnumType.cs:7]: The compiler warnings here are correct, let's implement `GetHashCode` as well 😅

File: Cesium.CodeGen/Ir/Types/StructType.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 namespace Cesium.CodeGen.Ir.Types;
 
-internal sealed class StructType : IGeneratedType
+internal sealed class StructType : IGeneratedType, IEquatable<StructType>
```
Comment (ForNeVeR) [inline Cesium.CodeGen/Ir/Types/StructType.cs:10]: Let's implement `GetHashCode` here, too.

**Review by ForNeVeR — CHANGES_REQUESTED** [review]

**Review by ForNeVeR — APPROVED** [review]
Thanks!

**Pull request merged**